prdoc/pr_7592.prdoc

```diff
@@ -0,0 +1,23 @@
+title: Add `Paras` authorize code_hash for `(force_)set_current_code` feature
+doc:
+- audience: Runtime Dev
+  description: |-
+    Closes: https://github.com/paritytech/polkadot-sdk/issues/7574
+    Relates to: https://github.com/paritytech/polkadot-sdk/issues/7591
+
+    This feature can be useful when we want to trigger `Paras::force_set_current_code(para, code)` from a different chain than the one where the `Paras` pallet is deployed.
+
+    The main reason is to avoid transferring the entire `new_code` wasm blob between chains.
+    Instead, we authorize `new_code_hash` with `root`, which can later be applied by `Paras::apply_authorized_force_set_current_code(para, new_code)` by anyone.
+
+
+    ## Open questions
+
+    - [ ] Do we need something like `poke_authorized_code_hash`? E.g. in case that we authorize code hash, but nobody would apply it and the parachain starts working with old code? Is this possible?
```

Comment:
```suggestion
    This feature can be useful when we want to trigger `Paras::force_set_current_code(para, code)` from a different chain than the one where the `Paras` pallet is deployed.

    The main reason is to avoid transferring the entire `new_code` wasm blob between chains.
    Instead, we authorize `new_code_hash` with `root`, which can later be applied by `Paras::apply_authorized_force_set_current_code(para, new_code)` by anyone.
```